frame/utility/src/lib.rs

```diff
@@ -478,3 +562,25 @@ impl<T: Config> Pallet<T> {
 			.expect("infinite length input; no invalid inputs for type; qed")
 	}
 }
+
+#[derive(Encode, Decode, Clone, PartialEq, RuntimeDebug, TypeInfo)]
+#[codec(dumb_trait_bound)]
+pub struct ForwardedCall<Nonce, BlockNumber, Call, AccountId> {
+	pub nonce: Nonce,
+	pub deadline: BlockNumber,
+	pub call: Box<Call>,
+	pub signer: MultiSigner,
```

Comment:
Why is the signer not generic?

Reply:
Where do we provide a generic signer? AFAIK MultiSigner was the closest we got?

Reply:
```
trait Config {
    type Signature: Verify<Signer = Self::Signer>;
    type Signer: IdentifyAccount<AccountId = <Self as frame_system::Config>::AccountId;
}
```

Reply:
Basically how I explained it on [SO](https://substrate.stackexchange.com/questions/305/how-to-verify-a-signature-in-a-pallet).  
Full pallet example [here](https://github.com/perun-network/perun-polkadot-pallet/blob/b5a1897cdf5c7755cddfb2b1841331acab349aba/src/lib.rs#L98) and [here](https://github.com/perun-network/perun-polkadot-pallet/blob/b5a1897cdf5c7755cddfb2b1841331acab349aba/src/types.rs#L177).  
This manual sig verification stuff is a bit tricky.